static/js/settings_users.js

```diff
@@ -559,6 +561,36 @@ export function show_edit_user_info_modal(user_id, from_user_info_popover, statu
             true,
             false,
         );
+
+        $("#edit-user-form").on("click", ".deactivate_user_button", (e) => {
+            e.preventDefault();
+            e.stopPropagation();
+            const user_id = $("#edit-user-form").data("user-id");
+            function handle_confirm() {
+                const url = "/json/users/" + encodeURIComponent(user_id);
+                channel.del({
+                    url,
+                    success() {
+                        dialog_widget.close_modal();
+                        if (!from_user_info_popover) {
+                            const row = get_user_info_row(user_id);
+                            update_view_on_deactivate(row);
+                        }
```

Comment:
I think the `update_view_on_deactivate` calls could be eliminated from these `success` handlers entirely if we just modify `user_events.js` to call `update_user_data` when changing the `active` flag on a user; this would be a substantial cleanup.

Also, while we're at this, I noticed that a user deactivated via this flow does not disappear from presence immediately; likely another thing that should be happening in the `user_events.js` handler block for a deactivated user.